src/arch_mips.py

```diff
@@ -667,14 +727,138 @@ def parse(
+                assert len(args) == 1
+                inputs = [Register("condition_bit")]
+            else:
+                assert len(args) == 2 and isinstance(args[0], Register)
+                inputs = [args[0]]
             jump_target = get_jump_target(args[-1])
             has_delay_slot = True
             is_conditional = True
+        elif mnemonic in cls.instrs_no_dest:
+            assert not any(isinstance(a, AsmAddressMode) for a in args)
+            inputs = [r for r in args if isinstance(r, Register)]
+        elif mnemonic in cls.instrs_store:
+            assert isinstance(args[0], Register)
+            inputs = [args[0]]
+            outputs = [make_memory_access(args[1])]
+            if isinstance(args[1], AsmAddressMode):
+                inputs.append(args[1].rhs)
+            if mnemonic == "sdc1":
+                inputs.append(args[0].other_f64_reg())
+        elif mnemonic in cls.instrs_source_first:
+            assert (
+                len(args) == 2
+                and isinstance(args[0], Register)
+                and isinstance(args[1], Register)
+            )
+            inputs = [args[0]]
+            outputs = [args[1]]
+        elif mnemonic in cls.instrs_destination_first:
+            assert isinstance(args[0], Register)
+            outputs = [args[0]]
+            mn_parts = mnemonic.split(".")
+            if mnemonic in (
+                "add.d",
+                "sub.d",
+                "neg.d",
+                "abs.d",
+                "sqrt.d",
+                "div.d",
+                "mul.d",
+                "mov.d",
+            ):
+                # f64 arithmetic operations; all registers are f64's
+                assert 2 <= len(args) <= 3
+                for reg in args[1:]:
+                    assert isinstance(reg, Register)
+                    inputs.extend([reg, reg.other_f64_reg()])
+                outputs.append(args[0].other_f64_reg())
+            elif mn_parts[0] in ("cvt", "trunc"):
+                # f64 conversion; either the input or output will be an f64
+                assert len(args) == 2 and isinstance(args[1], Register)
+                if mn_parts[2] == "d":
+                    inputs = [args[1], args[1].other_f64_reg()]
+                else:
+                    inputs = [args[1]]
+                if mn_parts[1] == "d":
+                    outputs.append(args[0].other_f64_reg())
+            elif mnemonic.startswith("l") and size is not None:
+                # Load instructions
+                assert len(args) == 2
+                inputs = [make_memory_access(args[1])]
+                if isinstance(args[1], AsmAddressMode):
+                    inputs.append(args[1].rhs)
+                if mnemonic in ("lwr", "lwl"):
+                    inputs.append(args[0])
+                elif mnemonic == "ldc1":
+                    outputs.append(args[0].other_f64_reg())
+            elif mnemonic == "la" and isinstance(args[1], AsmAddressMode):
```

Comment:
I think la can only take a symbol or (symbol + offset) as argument

Reply:
There's a specific case for this in the `ida/handwritten.s` test. It has `la $v0, 4($v0)`:

```asm
glabel test
la $v0, symbol
lw $v0, ($v0)
la $v1, symbol + 8
lw $v1, 4($v1)

# same as addition
la $v0, 4($v0)
...
```